### PR TITLE
fix: [PL-27875]: hide delegates list until user clicks on “Use only delegates with the following tags”

### PR DESCRIPTION
**Pull request by douglasnfrazao**
### Summary

Hide delegates list until the user selects "Use only delegates with the following tags”

#### Screenshots

BEFORE:
Captura de Tela 2022-12-01 às 12 16 19

AFTER:
Captura de Tela 2022-12-14 às 08 43 07

Captura de Tela 2022-12-01 às 12 13 56

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by douglasnfrazao** [comment]
retrigger titlecheck

**Review by tan-nhu — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Review by kapilkumawat86 — CHANGES_REQUESTED** [review]
Please check again issues days hide delegate list to use selects the use delegate, @abhinavrastogi-harness

**Review by abhinavrastogi-harness — COMMENTED** [review]
1. Can we hide the header of the table as well when the table is not being shown
2. We should add tests for the case when the table is visible

**Comment by douglasnfrazao** [comment]
@abhinavrastogi-harness the PR was updated removing the table header and creating tests

**Review by abhinavrastogi-harness — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Comment by douglasnfrazao** [comment]
retrigger jest

**Comment by douglasnfrazao** [comment]
retrigger coverage

**Review by kapilkumawat86 — APPROVED** [review]

File: src/modules/35-connectors/components/CreateConnector/commonSteps/DelegateSelectorStep/DelegateSelector/DelegateSelector.tsx [hunk 1]
```diff
@@ -318,24 +318,28 @@ export const DelegateSelector: React.FC<DelegateSelectorProps> = props => {
...
-          data-name="installNewDelegateButton"
-        >
-          {getString('connectors.testConnectionStep.installNewDelegate')}
-        </RbacButton>
-      </Layout.Horizontal>
-      <DelegateSelectorTable {...delegateSelectorTableProps} />
+      {mode !== DelegateOptions.DelegateOptionsAny && (
+        <>
+          <Layout.Horizontal flex={{ justifyContent: 'space-between' }} margin={{ bottom: 'medium' }}>
+            <Text font={{ size: 'medium', weight: 'semi-bold' }} color={Color.BLACK}>
```
Comment (Kira-Jennings) [inline src/modules/35-connectors/components/CreateConnector/commonSteps/DelegateSelectorStep/DelegateSelector/DelegateSelector.tsx:324]: Could a font variation be used here?

**Review by Kira-Jennings — APPROVED** [review]

**Pull request merged**